test/jdk/java/net/httpclient/http2/UserInfoTest.java

```diff
@@ -0,0 +1,111 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.net.SimpleSSLContext;
+import jdk.test.lib.net.URIBuilder;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+/**
+ * @test
+ * @bug 8292876
+ * @library /test/lib server
+ * @modules java.base/sun.net.www.http
+ *          java.net.http/jdk.internal.net.http.common
+ *          java.net.http/jdk.internal.net.http.frame
+ *          java.net.http/jdk.internal.net.http.hpack
+ * @run junit UserInfoTest
+ */
+
+public class UserInfoTest {
+
+    static class Http2TestHandler implements Http2Handler {
+        @Override
+        public void handle(Http2TestExchange e) throws IOException {
+            InputStream is = e.getRequestBody();
+            StringBuilder sb = new StringBuilder();
+            for (int ch; (ch = is.read()) != -1; ) {
+                sb.append((char) ch);
+            }
+            is.readAllBytes();
+            is.close();
+            if (e.getRequestURI().getAuthority().contains("user@")) {
+                e.sendResponseHeaders(500, -1);
+            } else {
+                e.sendResponseHeaders(200, -1);
+            }
+        }
+    }
+
+    private static Http2TestServer createServer(SSLContext sslContext) throws Exception {
+        Http2TestServer http2TestServer = new Http2TestServer("localhost", true, sslContext);
+        Http2TestHandler handler = new Http2TestHandler();
+        http2TestServer.addHandler(handler, "/");
+        return http2TestServer;
+    }
+
+    @Test
+    public void main() throws Exception {
+        SSLContext sslContext = new SimpleSSLContext().get();
+        Http2TestServer server = createServer(sslContext);
+        int port = server.getAddress().getPort();
+
+        server.start();
+
+        HttpClient client = HttpClient
+                .newBuilder()
+                .proxy(HttpClient.Builder.NO_PROXY)
+                .sslContext(sslContext)
+                .build();
+
+        URI uri = URIBuilder.newBuilder()
+                .scheme("https")
+                .userInfo("user")
+                .loopback()
+                .port(port)
+                .build();
+
+        HttpRequest request = HttpRequest
+                .newBuilder(uri)
+                .GET()
+                .build();
+
+        try {
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+            assertEquals(200, response.statusCode(), "Test Failed : " + response.uri().getAuthority());
```

Comment:
I think the `response.uri().getAuthority()` is not necessary in the message and perhaps could be confusing. But it's OK with me if you want to use it in the message.